build_tools/third_party/stblib/stb_image.c

```diff
@@ -0,0 +1,4 @@
+// By defining STB_IMAGE_IMPLEMENTATION the included header file will also
+// define the implementation.
+#define STB_IMAGE_IMPLEMENTATION
+#include "stb_image.h"
```

Comment:
Still add build_tools/third_party/stblib/ directory and this file so whoever uses stb_image.h doesn't need to worry about adding the MACRO or if it is added already. Also, if we decide to support bazel anyway this will be part of the compatibility work.